src/dependency-manager/src/spec/component-spec.ts

```diff
@@ -194,14 +194,14 @@ export class ComponentSpec {
   @JSONSchema({
     type: 'object',
     patternProperties: {
-      [Slugs.ComponentParameterValidator.source]: AnyOf(OutputDefinitionSpec),
+      [Slugs.ComponentParameterValidator.source]: AnyOf('string', 'number', 'boolean', OutputDefinitionSpec, 'null'),
```

Comment:
Wanted to avoid failing validation if the parameter is null which is valid and then fail on the outputs. Possible you want conditional behavior where in some scenarios you output a value, but in others it isn't set. Then the consumer behaves differently.